sdk/internal/recording/recording.go

```diff
@@ -85,340 +79,53 @@ const (
 	Secret_Base64String VariableType = "secret_base64String"
 )
 
-// defaultSanitizersSet tracks whether default sanitizers have been added (this would be neater with Go 1.19's atomic.Bool)
-var defaultSanitizersSet int32
+var (
+	// defaultSanitizersSet tracks whether default sanitizers have been added (this would be neater with Go 1.19's atomic.Bool)
+	defaultSanitizersSet int32
+
+	errUnsupportedAPI = errors.New("the vcr based recording API isn't supported. Use the test proxy instead")
+)
 
 // NewRecording initializes a new Recording instance
-//
-// Deprecated: call [Start] instead
 func NewRecording(c TestContext, mode RecordMode) (*Recording, error) {
-	// create recorder based on the test name, recordMode, variables, and sanitizers
-	recPath, varPath := getFilePaths(c.Name())
-	rec, err := recorder.NewAsMode(recPath, modeMap[mode], nil)
-	if err != nil {
-		return nil, err
-	}
-
-	// If the mode is set in the environment, let that override the requested mode
-	// This is to enable support for nightly live test pipelines
-	envMode := getOptionalEnv(ModeEnvironmentVariableName, string(mode))
-	mode = RecordMode(*envMode)
-
-	// initialize the Recording
-	recording := &Recording{
-		SessionName:              recPath,
-		RecordingFile:            recPath + ".yaml",
-		VariablesFile:            varPath,
-		Mode:                     mode,
-		variables:                make(map[string]*string),
-		previousSessionVariables: make(map[string]*string),
-		recorder:                 rec,
-		c:                        c,
-	}
-
-	// Try loading the recording if it already exists to hydrate the variables
-	err = recording.initVariables()
-	if err != nil {
-		return nil, err
-	}
-
-	// set the recorder Matcher
-	recording.Matcher = defaultMatcher(c)
-	rec.SetMatcher(recording.matchRequest)
-
-	// wire up the sanitizer
-	recording.Sanitizer = defaultSanitizer(rec)
-
-	return recording, err
+	return nil, errUnsupportedAPI
```

Comment:
This is the internal module we own and only we can bind to and call. We should remove the APIs entirely.

Reply:
That's a breaking change, and I'm worried there will be some corner case that causes builds to break (@chlowell and I extensively discussed this).

Reply:
I'll also add that we would never introduce breaking changes in production code outside of a major version change. We should hold our internal tooling to the same bar.